### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by Midnighter**
These are my inline comments and suggestions for changes. Comments are not rendered so please take a careful look at the diff.

### CODE REVIEW

File: taxprofiler-manuscript.qmd [hunk 1]
```diff
@@ -129,111 +129,117 @@ A common approach to address the differences and biases between the many differe
...
 
 ### Profiling
 
-There are many types of metagenomic profiling techniques, from profiling against whole-genome references with alignment or k-mer based approaches, to methods involving alignment to species-specific marker-gene families [@Quince2017-bl,@Ye2019-fl].
+There are many types of metagenomic profiling techniques, from profiling against whole-genome references with alignment or k-mer based approaches, to methods involving alignment to species-specific marker-gene families [@Quince2017-bl;@Ye2019-fl].
 nf-core/taxprofiler aims to support and include all established classification or profiling tools as requested by the community.
+<!--# (Moritz): I was actually wondering about this the other day. I think there are two possible approaches: 1. We add all tools on a per request basis as stated here. Upside: Everybody's happy. Downside: Maintenance becomes ever more burdensome over time. Do we ever remove profilers? 2. The taxprofiler maintainers choose the "best" representative tool within a specific category. -->
```
Comment (sofstam) [inline taxprofiler-manuscript.qmd:243]: I think it would be good to be able to remove profilers in the future.

Comment (jfy133) [reply taxprofiler-manuscript.qmd:243]: Why would maintainence become more burdensome over time? They should be fixed with the containers etc., no?

I would just keep adding as people ask. I 'm not sure we would have enough expertise to pick the 'best' representative tool for every discipline...

File: taxprofiler-manuscript.qmd [hunk 2]
```diff
@@ -129,111 +129,117 @@ A common approach to address the differences and biases between the many differe
...
+The fixed set of software are controlled through the use of the conda package manager or containers (e.g., Docker, Apptainer [previously known as Singularity]) from the stable Bioconda [@Gruning2018-vr] or BioContainers [@Da_Veiga_Leprevost2017-gl] repositories.
+This, coupled with the intrinsic Nextflow ability to execute on most infrastructure whether that is a local laptop (resource requirements permitting), traditional HPC, as well across common cloud providers also makes nf-core/taxprofiler a very portable pipeline that can be used in many contexts.
 
 ### Preprocessing
 
-Preprocessing steps in nf-core/taxprofiler are aimed at removing laboratory and sequencing artefacts that may influence taxonomically profiling, either for computing resource consumption and/or false-positive or false-negative classification reasons.
+Preprocessing steps in nf-core/taxprofiler are aimed at removing laboratory and sequencing artefacts that may influence taxonomic profiling, either for computing resource consumption or binary classification reasons.
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd]:
```suggestion
Preprocessing steps in nf-core/taxprofiler are aimed at removing laboratory and sequencing artefacts that may influence taxonomic profiling, either for computing resource consumption or and/or false-positive or false-negative classification reasons.
```

I know what I wrote was ugly but I don't have a better way of describing this atm (I wouldn't know what you meant here by 'binary classification')

File: taxprofiler-manuscript.qmd [hunk 3]
```diff
@@ -129,111 +129,117 @@ A common approach to address the differences and biases between the many differe
...
-For both short- and long-reads FastQC or Falco is run again to allow assessment on the performance of the adapter removal and/or pair-merging step.
-
-Low complexity sequences, e.g. sequences containing long stretches mono- or di-nucleotide repeats provide little specific genetic information that contribute to taxonomic identification, as they can align to many different reference genomes [@Schmieder2011-ea;@Clarke2019-al].
-Including such reads during taxonomic profiling can increase run-time and memory usage for little gain, as during lowest-common-ancestor (LCA) classification steps they will be assigned to high-level taxonomic levels (e.g. Kingdom).
+Falco was included for reduced memory requirements, in particular for long read sequencing data.
+Artificial library adapter sequences added during sequencing reduce alignment accuracy by reducing sequence specificity, and in some cases, may result in false-positive hits due to adapter sequence contamination in reference genomes [@Schaffer2018-be;@Breitwieser2018-xg] [^1].
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd]:
```suggestion
Artificial library adapter sequences added during sequencing reduce sequencing matching accuracy by reducing sequence specificity, and in some cases, may result in false-positive hits due to adapter sequence contamination in reference genomes [@Schaffer2018-be;@Breitwieser2018-xg] [^1].
```

File: taxprofiler-manuscript.qmd [hunk 4]
```diff
@@ -129,111 +129,117 @@ A common approach to address the differences and biases between the many differe
...
+Including such reads during taxonomic profiling can increase run-time and memory usage for little gain, as during lowest-common-ancestor (LCA) classification steps they will be assigned to high-level taxonomic ranks (e.g. Kingdom).
 nf-core/taxprofiler performs removal of these reads through complexity filtering algorithms as provided by fastp, BBDuk [@Bushnell2022-pf], or PRINSEQ++ [@Cantu2019-lh].
-Long read sequences often do not have such reads as lengths are sufficient enough to capture greater sequence diversity - but it is sometimes desired to only classify reads longer than a certain length - as these provide more precise taxonomic information [@Dilthey2019-ps;@Portik2022-jp].
+Long-read sequences often do not have such reads, as lengths are sufficient enough to capture greater sequence diversity - but it is sometimes desirable to only classify reads longer than a certain length - as these provide more precise taxonomic information [@Dilthey2019-ps;@Portik2022-jp].
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd]:
```suggestion
Long read sequences often do not have such reads, as lengths are sufficient enough to capture greater sequence diversity - but it is sometimes desirable to only classify reads longer than a certain length - as these provide more precise taxonomic information [@Dilthey2019-ps;@Portik2022-jp].
```

File: taxprofiler-manuscript.qmd [hunk 5]
```diff
@@ -129,111 +129,117 @@ A common approach to address the differences and biases between the many differe
...
+Short read host removal can be carried out with Bowtie2 [@Langmead2012-ik;@Langmead2019-ej] and minimap2 [@Li2018-qq] for long reads, both in combination with SAMtools [@Li2009-wy;@Danecek2021-gj], where reads are aligned against the reference genome and the off-target (unaligned) reads are then converted back to FASTQ format for classification.
 
 Finally, nf-core/taxprofiler can optionally perform run merging where libraries have been sequenced over multiple lanes to generate one profile per sample or library.
-The final set of reads used for profiling can be optionally saved for downstream re-use.
-
-Throughout all steps, relevant statistics and log files are generated and used both for the final pipeline run report as well as saved into the results directory of the pipeline run for further inspection where necessary.
+The final set of reads used for profiling can be optionally saved for downstream re-use. Throughout all steps, relevant statistics and log files are generated and used both for the final pipeline run report as well as saved into the results directory of the pipeline run for further inspection where necessary.
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd]:
```suggestion
The final set of reads used for profiling can be optionally saved for downstream re-use.
Throughout all steps, relevant statistics and log files are generated and used both for the final pipeline run report as well as saved into the results directory of the pipeline run for further inspection where necessary.
```

File: taxprofiler-manuscript.qmd [hunk 6]
```diff
@@ -269,27 +276,27 @@ Users can also optionally request output of per-read classification output, and
...
-To facilitate more user-friendly cross-comparison between tools, nf-core/taxprofiler utilises the TAXPASTA tool [@Beber2023-hk] to generate standardised profiles and generate multi-sample tables.
+One of the challenges that researchers face when comparing multiple taxonomic classifiers or profilers is the heterogenous output formats that are produced, that often require custom parsing and merging scripts for each tool to standardise.
+To facilitate more user-friendly cross-comparisons between tools, nf-core/taxprofiler utilises the TAXPASTA tool [@Beber2023-hk] to generate standardised profiles and generate multi-sample tables.
 
-Summary statistics for the entire pipeline are visualised and displayed in a customisable MultiQC report [@Ewels2020-vi], when supported, to support user quality control of data and pipeline runs.
+Summary statistics for the entire pipeline are visualised and displayed in a customisable MultiQC report [@Ewels2020-vi]. When supported, quality control of data and pipeline runs are shown for manual verification.
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd]:
```suggestion
Summary statistics for the entire pipeline are visualised and displayed in a customisable MultiQC report [@Ewels2020-vi].
When supported, quality control of data and pipeline runs are shown for manual verification.
```

File: taxprofiler-manuscript.qmd [hunk 7]
```diff
@@ -310,70 +317,79 @@ Run metric can be seen in Figure XXXXXX .
 
 ### Comparison with other solutions
 
-nf-core/taxprofiler has been specifically developed for the analysis of _metagenomic_ data.
-While other types of taxonomic profiling data such as 16S amplicon sequencing are well established fields with a range of popular high-quality and best-practise tools pipelines (e.g. [@Blanco-Miguez2023-cq;@Schloss2009-vk]) and databases [@DeSantis2006-cm;@Yilmaz2014-js], whereas 'gold standard' tools and databases for metagenomics remains are much less established - thus the need for highly-multiplexed classification is more desirable for the newer metagenomics methods.
+nf-core/taxprofiler has been specifically developed for the analysis of whole-genome, _metagenomic_ sequencing data.
+While other types of taxonomic profiling data such as 16S amplicon sequencing are well established fields with a range of popular high-quality and best-practise tools pipelines (e.g. [@Blanco-Miguez2023-cq;@Schloss2009-vk]) and databases [@DeSantis2006-cm;@Yilmaz2014-js], 'gold standard' tools and databases for metagenomics remain much less established. Thus, the need for highly-multiplexed classification is more desirable for the newer metagenomics methods.
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd]:
```suggestion
While other types of taxonomic profiling data such as 16S amplicon sequencing are well established fields with a range of popular high-quality and best-practise tools pipelines (e.g. [@Blanco-Miguez2023-cq;@Schloss2009-vk]) and databases [@DeSantis2006-cm;@Yilmaz2014-js], 'gold standard' tools and databases for metagenomics remain much less established.
Thus, the need for highly-multiplexed classification is more desirable for the newer metagenomics methods.
```

File: taxprofiler-manuscript.qmd [hunk 8]
```diff
@@ -310,70 +317,79 @@ Run metric can be seen in Figure XXXXXX .
...
 
-Criteria covering portability also overlaps with accessibility, as it implies the options and ease that different users can run on different types of computing infrastructure, whether that is on their own laptop, on a HPC cluster or in the cloud.
+Criteria covering portability also overlap with accessibility, as it implies options for and ease of different users running on different types of computing infrastructure, whether that is on their own laptop, on an HPC cluster, or in the cloud.
 Unipro UGENE is the only pipeline that supports execution on all three major operating systems (Linux, OSX, Windows), whereas StaG-mwc and nf-core/taxprofiler can be run on unix operating systems, and sunbeam and TAMA are only being supported on Linux.
+<!--# (Moritz): Is this so? Nextflow requires bash, but there is git bash, anaconda terminal, and the WSL for Windows. So this seems a bit strict? -->
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd:340]: Does WSL work reliably though and that is sort of technically still 'linux' (even if _within_ Windows)? I've seen varying reports... I was basing it on what had been reported

File: taxprofiler-manuscript.qmd [hunk 9]
```diff
@@ -310,70 +317,79 @@ Run metric can be seen in Figure XXXXXX .
...
-Most pipelines support some form of host removal (only TAMA did not support this), and it is likely possible with Unipro UGENE through user customisation of the workflow. In all cases, this consists of mapping processed reads with an aligner and taking the off-target reads (as implemented in nf-core/taxprofiler), however StaG-mwc has an additional separate metagenomic host removal step with Kraken2.
+However, installing software manually across different infrastructures can result in variability in the execution of each software [^2] [@Di_Tommaso2017-xu].The current most popular solution to the problem of inconsistent software environments is to use container engines such as Docker or Apptainer to run container images which are isolated, deterministic computing environments which can be executed by any system providing a container runtime.
+Only Unipro UGENE does not document the use of a container system with nf-core/taxprofiler offering the biggest choice for users courtesy of Nextflow (6 different engine systems at the time of writing).
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd]:
```suggestion
Only Unipro UGENE does not document the use of a container system, with nf-core/taxprofiler offering the biggest choice for users courtesy of Nextflow (6 different engine systems at the time of writing).
```

File: taxprofiler-manuscript.qmd [hunk 10]
```diff
@@ -310,70 +317,79 @@ Run metric can be seen in Figure XXXXXX .
...
+Another motivation for developing nf-core/taxprofiler, despite the large number of existing metagenomics pipelines is that by establishing a taxonomic profiling pipeline within the nf-core ecosystem, it is possible to begin building both standalone but also an integrated suite of powerful interconnected pipelines for the major stages of metagenomic workflows.
 Existing microbial- and metagenomics- related pipelines within the nf-core initiative include nf-core/ampliseq, nf-core/mag, and nf-core/funcscan.
-We expect over time the ability to link inputs and outputs of each workflow to develop comphrensive metagenomic analyses, while still maintaing powerful standalone pipelines, providing maximal user choice.
+We expect over time the ability to link inputs and outputs of each workflow to develop comprehensive metagenomic analyses, while still maintaing powerful standalone pipelines, providing maximal user choice.
```
Comment (jfy133) [inline taxprofiler-manuscript.qmd]:
```suggestion
We expect over time the ability to link inputs and outputs of each workflow to develop comprehensive metagenomic analyses, while still maintaining powerful standalone pipelines, providing maximal user choice.
```

**Review by jfy133 — APPROVED** [review]
In this case I think it will be easier to merge in the changes and then address the comments in a separate PR where necessary

File: taxprofiler-manuscript.qmd (hunk 1 above)
Comment (Midnighter) [reply taxprofiler-manuscript.qmd:243]: You have more and more modules whose versions need to be updated and potentially arguments need to be fixed. Additionally, the pipeline logic itself keeps growing and becomes harder to grok.

File: taxprofiler-manuscript.qmd (hunk 8 above)
Comment (Midnighter) [reply taxprofiler-manuscript.qmd:340]: I'm not a user of the WSL myself, so I don't know.

**Pull request merged**